main.go

```diff
@@ -175,10 +176,21 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		obj.SetNamespace(req.Namespace)
 		// don't requeue after deletion.
 		now := metav1.Now()
-		return ctrl.Result{}, r.store.Upsert(ctx, obj, r.gvk.PreferredVersion, r.orgID, &now)
+		requestID, err := r.store.Upsert(ctx, obj, r.gvk.PreferredVersion, r.orgID, &now)
+		if err != nil {
+			log.WithValues("request_id", requestID).
+				Error(err, "could not publish to store")
+		}
+		return ctrl.Result{}, err
+	}
+
+	requestID, err := r.store.Upsert(ctx, obj, r.gvk.PreferredVersion, r.orgID, nil)
+	if err != nil {
+		log.WithValues("request_id", requestID).
+			Error(err, "could not publish to store")
```

Comment:
Can we generate the request ID at the start of the reconciliation loop, and ensure it's present on all relevant logs? e.g. as a field on https://github.com/snyk/kubernetes-scanner/blob/c75fbad3357e56e5968f44e06cfc0f2e5d597fa3/main.go#L158

It would therefore be passed into Upsert(), rather than generated there.